pkg/client/client.go

```diff
@@ -517,6 +521,46 @@ func (w *uploadChunkWorker) Start() <-chan error {
 }
 
 func (w *uploadChunkWorker) upload(job uploadChunkJob) error {
+	// return errors.New("abc")
+	pipeReader, pipeWriter := io.Pipe()
+	writer := multipart.NewWriter(pipeWriter)
+	go func() {
+		defer pipeWriter.Close()
+		defer writer.Close()
+		if err := writeMultipartRequest(writer, job); err != nil {
+			fmt.Fprintf(w.DumpOut, "Error writing multipart request %v", err)
+		}
+	}()
+	// client trace to log whether the request's underlying tcp connection was re-used
+	clientTrace := &httptrace.ClientTrace{
+		GotConn: func(info httptrace.GotConnInfo) {
+			if !info.Reused {
+				const msg = "tcp connection was not reused uploading file chunk: %q," +
+					"chunk number: %d, chunk total: %d\n"
+				fmt.Fprintf(w.DumpOut, msg,
+					filepath.Base(job.Filename), job.ChunkNumber, job.TotalChunks)
+			}
+		},
+	}
+	traceCtx := httptrace.WithClientTrace(context.Background(), clientTrace)
```

Comment:
Use the same context used for the workers instead of `context.Background()`: Cancelling the workers should also cancel the request if it's already started.

Reply:
used the per request context.